src/nostrdb.c

```
@@ -6783,6 +6866,138 @@ int ndb_process_giftwrap(secp256k1_context *secp,
+		/* recalculate id */
+		if (!ndb_calculate_id(inner,
+				      inner_scratch + note_size,
+				      inner_scratch_size - note_size,
+				      ndb_note_id(inner)))
+			return 0;
+
+		flags = ndb_note_flags(inner);
+		*flags = *flags | NDB_NOTE_FLAG_RUMOR;
+
+		inner_msg = malloc(note_size);
+		memcpy(inner_msg, inner, note_size);
```

Cppcheck (2.19.0)

[warning] If memory allocation fails, then there is a possible null pointer dereference

(nullPointerOutOfMemory)